### PR TITLE
[Partial] Build wasi-libc for Windows

### PR DESCRIPTION
**Pull request by loganek**
Hi,
Our team is planning to build wasi-libc library on Windows (we can't use libuv). This PR is a first step to make it working, but there's still a number of changes to get it fully working.

Because we expect the changes in total will be significant, could we have a `dev/wasi-libc-windows` branch and incrementally add changes there to simplify the review process?

### CODE REVIEW

**Comment by wenyongh** [comment]
> Hi, Our team is planning to build wasi-libc library on Windows (we can't use libuv). This PR is a first step to make it working, but there's still a number of changes to get it fully working.
> 
> Because we expect the changes in total will be significant, could we have a `dev/wasi-libc-windows` branch and incrementally add changes there to simplify the review process?

@loganek Sorry for late response, I have just created branch `dev/wasi-libc-windows`. And is this PR ready to review and merge?

**Comment by loganek** [comment]
Thanks a lot @wenyongh . I just updated the PR to point to the `dev/wasi-libc-windows` branch. Yes, it's ready for review.

File: core/shared/platform/linux-sgx/sgx_thread.c [hunk 1]
```diff
@@ -210,3 +210,58 @@ os_thread_get_stack_boundary()
...
+
+int
+os_rwlock_init(korp_rwlock *lock)
+{
+    assert(lock);
+
+    if (pthread_rwlock_init(lock, NULL) != BHT_OK)
+        return BHT_ERROR;
+
+    return BHT_OK;
```
Comment (wenyongh) [inline core/shared/platform/linux-sgx/sgx_thread.c:222]: Had better use the macro `SGX_DISABLE_PTHREAD` to control the code, see other functions above:
```C
#ifndef SGX_DISABLE_PTHREAD
    assert(lock);

    if (pthread_rwlock_init(lock, NULL) != BHT_OK)
        return BHT_ERROR;
#endif

    return BHT_OK;
```

File: core/shared/platform/linux-sgx/sgx_thread.c [hunk 2]
```diff
@@ -210,3 +210,58 @@ os_thread_get_stack_boundary()
...
+}
+
+int
+os_rwlock_rdlock(korp_rwlock *lock)
+{
+    assert(lock);
+
+    if (pthread_rwlock_rdlock(lock) != BHT_OK)
```
Comment (wenyongh) [inline core/shared/platform/linux-sgx/sgx_thread.c:230]: same as above

File: core/shared/platform/linux-sgx/sgx_thread.c (hunk 1 above)
Comment (loganek) [reply core/shared/platform/linux-sgx/sgx_thread.c]: updated

File: core/shared/platform/linux-sgx/sgx_thread.c (hunk 2 above)
Comment (loganek) [reply core/shared/platform/linux-sgx/sgx_thread.c:233]: updated

**Pull request merged**